okhttp/src/main/java/io/grpc/okhttp/OkHttpChannelBuilder.java

```diff
@@ -243,6 +245,27 @@ public final OkHttpChannelBuilder sslSocketFactory(SSLSocketFactory factory) {
   }
 
   /**
+   * Set the hostname verifier to use when using TLS negotiation. The hostnameVerifier is only used
+   * if an sslSocketFactory is set. If the hostname verifier is not set, a default hostname
+   * verifier is used.
+   *
+   * <p>Be careful when setting a custom hostname verifier! By setting a non-null value, you are
```

Comment:
Also add something like:

> `This method should not be used to avoid hostname verification, even during testing, since {@link #overrideAuthority} is a safer alternative as it does not disable any security checks.`